ruma-federation-api/src/keys/claim_keys/v1.rs

```diff
@@ -0,0 +1,38 @@
+//! [POST /_matrix/federation/v1/user/keys/claim](https://matrix.org/docs/spec/server_server/r0.1.4#id91)
+
+use std::collections::BTreeMap;
+
+use ruma_api::ruma_api;
+use ruma_identifiers::{DeviceId, DeviceKeyAlgorithm, DeviceKeyId, UserId};
+use serde::{Deserialize, Serialize};
+
+ruma_api! {
+    metadata: {
+        description: "Claims one-time keys for use in pre-key messages.",
+        method: POST,
+        name: "claim_keys",
+        path: "/_matrix/federation/v1/user/keys/claim",
+        rate_limited: false,
+        requires_authentication: false,
+    }
+
+    request: {
+        /// The keys to be claimed.
+        /// #[ruma_api(path)]
+        one_time_keys: BTreeMap<UserId, BTreeMap<Box<DeviceId>, DeviceKeyAlgorithm>>,
+    }
+
+    response: {
+        /// One-time keys for the queried devices
+        one_time_keys: BTreeMap<UserId, BTreeMap<Box<DeviceId>, BTreeMap<DeviceKeyId, KeyObject>>>
+    }
+}
```

Comment:
Again, constructors for `Request` and `Response` would be nice.

Reply:
The type of both the request's and response's `one_time_keys` member is quite long, should I add a type alias for both to avoid clutter and/or reduce likelihood of writing them wrong?

Reply:
Sounds good 👍